### ISSUE TITLE
Can't Login

### ISSUE THREAD

**Issue by pred11** [itzg/kidsbank-js#49]
Hello, I've got things downloaded, compiled and packaged into a JAR file.  Mongo is setup, running and populated with the initial tables, certificates are applied, Facebook app created; however, I cannot login.  I don't get a redirect to Facebook to authorize the application which I think is the first step.

```
2021-08-29 17:03:59.080  WARN 3545 --- [nio-8080-exec-1] m.i.kidsbank.web.SimpleErrorController   : Intercepted web controller error: request=ServletWebRequest: uri=/error;client=127.0.0.1;session=BF4B9DC05E6A24E15CC8B92A6AE204AA, error={timestamp=Sun Aug 29 17:03:59 CDT 2021, status=403, error=Forbidden, message=Access Denied, path=/oauth2/authorization/facebook}
2021-08-29 17:03:59.097 DEBUG 3545 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Exiting from "ERROR" dispatch, status 307
2021-08-29 17:03:59.105 DEBUG 3545 --- [nio-8080-exec-2] o.s.web.servlet.DispatcherServlet        : GET "/?error=Forbidden&message=Access%20Denied", parameters={masked}
2021-08-29 17:03:59.109 DEBUG 3545 --- [nio-8080-exec-2] o.s.web.servlet.DispatcherServlet        : "FORWARD" dispatch for GET "/index.html?error=Forbidden&message=Access%20Denied", parameters={masked}
```

**Comment by itzg** [comment]
At some point I was also having issues with Facebook integration, but I can't remember exactly what the symptoms were. I always use the GitHub integration now, so you may want to try that. I'll do some local testing of Facebook also.

**Comment by itzg** [comment]
I'm not sure what my previous issue was before since it now seems to be working fine for me.

What I usually do in these cases is enable Spring Security logging by setting the property `logging.level.org.springframework.security` to `debug`.

An additional way to debug the oauth handoff is with the backend running, start the UI/frontend, browse to <https://localhost:3000/> and open the Chrome Dev Tools (or equivalent). In the Network tab, enable "Preserve Log", as shown below, since that will keep the network activity across redirects.

![image](https://user-images.githubusercontent.com/988985/131276072-ff744d46-73d9-4ce4-92a0-f9e2f19148e5.png)

Let me know what the debug logs from the Spring Boot app show and where the browser network activity seems to hit a deadend.

**Comment by pred11** [comment]
Thanks for the quick reply!  I did get it sorted out and running.  I did switch to the GitHub option as well.  I see Facebook has some other requirements I didn't want to deal with right now.

My issues was not including all of the required environment variables.  I then had an issue where the OAuth2 callback was incorrect.  I fixed the environment variables by looking at the k8s manifests you have and then the callback by using the Chrome Dev Tools and Spring documentation.

Here was the docker command I used:
``` docker run -e GITHUB_CLIENT_ID="Ivxxxxxxxxxxx84" -e GITHUB_CLIENT_SECRET="xxxxxxxxxxxxxxxxxxxxxxxxxx" -e SPRING_DATA_MONGODB_HOST=10.1.1.130 -e SPRING_PROFILES_ACTIVE=oauth2 -e SERVER_USE_FORWARD_HEADERS="true" -p 8080:8080 itzg/kidsbank-js:latest ```
